core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1415,14 +1420,16 @@ class ReplicaManager(val config: KafkaConfig,
                             correlationId: Int,
                             responseMap: mutable.Map[TopicPartition, Errors],
                             highWatermarkCheckpoints: OffsetCheckpoints) : Set[Partition] = {
+    val traceLoggingEnabled = stateChangeLogger.isTraceEnabled
+    stateChangeLogger.info(s"Handling LeaderAndIsr request correlationId $correlationId from controller $controllerId " +
```

Comment:
Let's just add an info level logging with partition count in the caller becomeLeaderOrFollower() instead of at the beginning of makeLeader() and makeFollowers(). This is because sometimes we ignore partitions in becomeLeaderOrFollower().